### ISSUE TITLE
Stopped working since a few days

### ISSUE THREAD

**Issue by metbril** [nimroddolev/chime_tts#151]
## System Information

version | core-2024.6.4
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.2
os_name | Linux
os_version | 6.6.33-haos
arch | aarch64
timezone | Europe/Amsterdam
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 4730
Installed Version | 1.34.0
Stage | running
Available Repositories | 1393
Downloaded Repositories | 44

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | 17 augustus 2024 om 02:00
relayer_connected | true
relayer_region | eu-central-1
remote_enabled | true
remote_connected | true
alexa_enabled | false
google_enabled | false
remote_server | eu-central-1-10.ui.nabu.casa
certificate_status | ready
instance_id | 15e19ba1b93a4cee9365912cf8ddfcdd
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 12.4
-- | --
update_channel | stable
supervisor_version | supervisor-2024.06.0
agent_version | 1.6.0
docker_version | 26.1.4
disk_total | 56.6 GB
disk_used | 11.5 GB
healthy | true
supported | true
host_connectivity | true
supervisor_connectivity | true
ntp_synchronized | true
virtualization | 
board | odroid-n2
supervisor_api | ok
version_api | ok
installed_addons | Samba share (12.3.1), Advanced SSH & Web Terminal (18.0.0), File editor (5.8.0), Home Assistant Google Drive Backup (0.112.1), Z-Wave JS (0.6.1), Mosquitto broker (6.4.1), Zigbee2MQTT (1.38.0-1), AdGuard Home (5.1.1), Studio Code Server (5.15.0)

</details>

<details><summary>Dashboards</summary>

dashboards | 8
-- | --
resources | 38
views | 38
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 15 juni 2024 om 09:32
-- | --
current_recorder_run | 22 juni 2024 om 20:27
estimated_db_size | 732.95 MiB
database_engine | sqlite
database_version | 3.44.2

</details>

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [previous issues](https://github.com/nimroddolev/chime_tts/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

```
2024-06-22 21:00:19.529 WARNING (MainThread) [homeassistant.util.loop] Detected blocking call to open inside the event loop by custom integration 'chime_tts' at custom_components/chime_tts/helpers/filesystem.py, line 119: audio.export(audio_full_path, format="mp3") (offender: /usr/local/lib/python3.12/site-packages/pydub/utils.py, line 60: fd = open(fd, mode=mode)), please create a bug report at https://github.com/nimroddolev/chime_tts/issues
```

For a few days now, the chime integration has stopped working. I don't know if its the upgrade from 1.0.0 to 1.1.0 and 1.1.2 or the upgrade from HA 2024.6.2 or .3 to 20204.6.4.

Testing the TTS in the browser directly is working just fine. The sentences are transformed to mp3 that is played just fine in-browser or on the media player.

But when I use the chime from the developer tools, I get a "green checkmark" but no sound at all.

Hope the provided info will be of help.

### Reproduction steps

1.
2.
3.
...

### Debug logs

```text
2024-06-22 21:00:19.287 DEBUG (MainThread) [custom_components.chime_tts] ----- Chime TTS Say Called. Version v1.0.0 -----
2024-06-22 21:00:19.287 DEBUG (MainThread) [custom_components.chime_tts.queue_manager] Adding service call to queue
2024-06-22 21:00:19.289 DEBUG (MainThread) [custom_components.chime_tts.helpers.media_player]  players found: [{'entity_id': 'media_player.living_speaker', 'should_change_volume': True, 'initial_volume_level': 0.06, 'group_members_supported': True, 'announce_supported': True, 'resume_media_player': False}]
2024-06-22 21:00:19.289 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] ----- General Parameters -----
2024-06-22 21:00:19.289 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers]  * entity_ids = ['media_player.living_speaker']
2024-06-22 21:00:19.289 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers]  * cache = False
2024-06-22 21:00:19.289 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers]  * offset = 450.0
2024-06-22 21:00:19.289 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers]  * final_delay = 0.0
2024-06-22 21:00:19.289 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers]  * message = Hallo.
2024-06-22 21:00:19.289 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers]  * language = nl-NL
2024-06-22 21:00:19.289 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers]  * tts_platform = cloud
2024-06-22 21:00:19.289 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers]  * tts_speed = 100.0
2024-06-22 21:00:19.289 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers]  * tts_pitch = 0
2024-06-22 21:00:19.290 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers]  * announce = False
2024-06-22 21:00:19.290 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers]  * volume_level = 0.3
2024-06-22 21:00:19.290 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers]  * join_players = False
2024-06-22 21:00:19.290 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers]  * unjoin_players = False
2024-06-22 21:00:19.290 ERROR (MainThread) [custom_components.chime_tts.helpers.helpers] An unexpected error occurred while parsing options YAML: 'dict' object has no attribute 'read'
2024-06-22 21:00:19.291 DEBUG (MainThread) [custom_components.chime_tts]  - Generating new TTS audio with parameters:
2024-06-22 21:00:19.291 DEBUG (MainThread) [custom_components.chime_tts]     * tts_platform = 'cloud'
2024-06-22 21:00:19.291 DEBUG (MainThread) [custom_components.chime_tts]     * message = 'Hallo.'
2024-06-22 21:00:19.291 DEBUG (MainThread) [custom_components.chime_tts]     * cache = False
2024-06-22 21:00:19.291 DEBUG (MainThread) [custom_components.chime_tts]     * language = "nl-NL"
2024-06-22 21:00:19.291 DEBUG (MainThread) [custom_components.chime_tts]     * options = {}
2024-06-22 21:00:19.521 DEBUG (MainThread) [custom_components.chime_tts]    ...TTS audio generated in 230.0ms
2024-06-22 21:00:19.521 DEBUG (MainThread) [custom_components.chime_tts]  - Saving mp3 file to folder: /media/sounds/temp/chime_tts/...
2024-06-22 21:00:19.529 WARNING (MainThread) [homeassistant.util.loop] Detected blocking call to open inside the event loop by custom integration 'chime_tts' at custom_components/chime_tts/helpers/filesystem.py, line 119: audio.export(audio_full_path, format="mp3") (offender: /usr/local/lib/python3.12/site-packages/pydub/utils.py, line 60: fd = open(fd, mode=mode)), please create a bug report at https://github.com/nimroddolev/chime_tts/issues
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module>
    sys.exit(main())
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main
    exit_code = runner.run(runtime_conf)
  File "/usr/src/homeassistant/homeassistant/runner.py", line 190, in run
    return loop.run_until_complete(setup_and_run_hass(runtime_config))
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 672, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 639, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1988, in _run_once
    handle._run()
  File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/config/custom_components/chime_tts/queue_manager.py", line 33, in async_process_queue
    result = await asyncio.wait_for(
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
  File "/config/custom_components/chime_tts/__init__.py", line 138, in async_say_execute
    audio_dict = await async_get_playback_audio_path(params, options)
  File "/config/custom_components/chime_tts/__init__.py", line 778, in async_get_playback_audio_path
    new_audio_file = filesystem_helper.save_audio_to_folder(output_audio, _data.get(initial_save_folder_key, None))
  File "/config/custom_components/chime_tts/helpers/filesystem.py", line 119, in save_audio_to_folder
    audio.export(audio_full_path, format="mp3")

2024-06-22 21:00:19.537 WARNING (MainThread) [homeassistant.util.loop] Detected blocking call to open inside the event loop by custom integration 'chime_tts' at custom_components/chime_tts/helpers/filesystem.py, line 119: audio.export(audio_full_path, format="mp3") (offender: /usr/local/lib/python3.12/site-packages/pydub/audio_segment.py, line 962: with open(os.devnull, 'rb') as devnull:), please create a bug report at https://github.com/nimroddolev/chime_tts/issues
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module>
    sys.exit(main())
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main
    exit_code = runner.run(runtime_conf)
  File "/usr/src/homeassistant/homeassistant/runner.py", line 190, in run
    return loop.run_until_complete(setup_and_run_hass(runtime_config))
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 672, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 639, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1988, in _run_once
    handle._run()
  File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/config/custom_components/chime_tts/queue_manager.py", line 33, in async_process_queue
    result = await asyncio.wait_for(
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
  File "/config/custom_components/chime_tts/__init__.py", line 138, in async_say_execute
    audio_dict = await async_get_playback_audio_path(params, options)
  File "/config/custom_components/chime_tts/__init__.py", line 778, in async_get_playback_audio_path
    new_audio_file = filesystem_helper.save_audio_to_folder(output_audio, _data.get(initial_save_folder_key, None))
  File "/config/custom_components/chime_tts/helpers/filesystem.py", line 119, in save_audio_to_folder
    audio.export(audio_full_path, format="mp3")

2024-06-22 21:00:19.667 DEBUG (MainThread) [custom_components.chime_tts.helpers.filesystem]  - File saved to path: /media/sounds/temp/chime_tts/rc6_ty4c.mp3
2024-06-22 21:00:19.675 WARNING (MainThread) [homeassistant.util.loop] Detected blocking call to open inside the event loop by custom integration 'chime_tts' at custom_components/chime_tts/__init__.py, line 794: duration = len(AudioSegment.from_file(new_audio_file)) / 1000.0 (offender: /usr/local/lib/python3.12/site-packages/pydub/utils.py, line 60: fd = open(fd, mode=mode)), please create a bug report at https://github.com/nimroddolev/chime_tts/issues
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module>
    sys.exit(main())
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main
    exit_code = runner.run(runtime_conf)
  File "/usr/src/homeassistant/homeassistant/runner.py", line 190, in run
    return loop.run_until_complete(setup_and_run_hass(runtime_config))
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 672, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 639, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1988, in _run_once
    handle._run()
  File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/config/custom_components/chime_tts/queue_manager.py", line 33, in async_process_queue
    result = await asyncio.wait_for(
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
  File "/config/custom_components/chime_tts/__init__.py", line 138, in async_say_execute
    audio_dict = await async_get_playback_audio_path(params, options)
  File "/config/custom_components/chime_tts/__init__.py", line 794, in async_get_playback_audio_path
    duration = len(AudioSegment.from_file(new_audio_file)) / 1000.0

2024-06-22 21:00:19.891 DEBUG (MainThread) [custom_components.chime_tts]  - Chime TTS audio generated:
2024-06-22 21:00:19.891 DEBUG (MainThread) [custom_components.chime_tts]    - audio_duration = 1.632
2024-06-22 21:00:19.891 DEBUG (MainThread) [custom_components.chime_tts]    - local_path = "/media/sounds/temp/chime_tts/rc6_ty4c.mp3"
2024-06-22 21:00:19.891 DEBUG (MainThread) [custom_components.chime_tts]    - public_path = None
2024-06-22 21:00:19.891 DEBUG (MainThread) [custom_components.chime_tts]    - media_content_id = "media-source://media_source/local/sounds/temp/chime_tts/rc6_ty4c.mp3"
2024-06-22 21:00:19.891 DEBUG (MainThread) [custom_components.chime_tts]  *** Pre-Playback Actions *** 
2024-06-22 21:00:19.892 DEBUG (MainThread) [custom_components.chime_tts]  - Setting volume level to 0.3 for 1 non-fade in/out media_player
2024-06-22 21:00:19.892 DEBUG (MainThread) [custom_components.chime_tts.helpers.media_player]  - Setting media_player.living_speaker's volume to 0.3
2024-06-22 21:00:19.914 DEBUG (MainThread) [custom_components.chime_tts]  *** Chime TTS playback ***
2024-06-22 21:00:19.914 DEBUG (MainThread) [custom_components.chime_tts]    1 Standard media player detected:
2024-06-22 21:00:19.914 DEBUG (MainThread) [custom_components.chime_tts]      - media_player.living_speaker
2024-06-22 21:00:19.914 DEBUG (MainThread) [custom_components.chime_tts]    Calling media_player.play_media with data:
2024-06-22 21:00:19.914 DEBUG (MainThread) [custom_components.chime_tts]      - media_content_type: music
2024-06-22 21:00:19.914 DEBUG (MainThread) [custom_components.chime_tts]      - media_content_id: media-source://media_source/local/sounds/temp/chime_tts/rc6_ty4c.mp3
2024-06-22 21:00:19.914 DEBUG (MainThread) [custom_components.chime_tts]      - entity_id: ['media_player.living_speaker']
2024-06-22 21:00:19.916 DEBUG (MainThread) [custom_components.chime_tts]  - Waiting 1.632s for audio playback to complete...
2024-06-22 21:00:21.553 DEBUG (MainThread) [custom_components.chime_tts.helpers.media_player]  - Waiting until 1 media_player isn't playing...
2024-06-22 21:00:21.553 DEBUG (MainThread) [custom_components.chime_tts.helpers.media_player]    ✔ media_player.living_speaker
2024-06-22 21:00:21.553 DEBUG (MainThread) [custom_components.chime_tts]  *** Post-Playback Actions ***
2024-06-22 21:00:21.554 DEBUG (MainThread) [custom_components.chime_tts.helpers.media_player]  - Setting media_player.living_speaker's volume to 0.06
2024-06-22 21:00:21.582 DEBUG (MainThread) [custom_components.chime_tts] ----- Chime TTS Say Completed in 2.29 s -----
```

**Comment by metbril** [comment]
I downgraded from 1.1.2 to 1.0.0 but it still does not work. Perhaps it's an incompatibility with the latest HA Core. I hesitate to roll back that upgrade, however.

**Comment by nimroddolev** [comment]
Hi @metbril. Sorry to hear you've been having this issue.

Your debug logs are from v1.0.0, which still have the blocking calls warning messages and other potential bugs and issues that have been resolved in v1.1.x. To help isolate the cause of your specific issue I would like ask you to please upgrade back to the latest stable version in order to rule out these issues. 

I would like to understand whether the issue is with the MP3 file generated by Chime TTS, or with the playback.

### 1. Test the generated MP3
To determine whether Chime TTS successfully generates the MP3 file for playback, please either:
1. Call the `chime_tts.say_url` service, or
2. Call the `chime_tts.say` service, with `cache` set to `true`
and then review the log messages to get the generated MP3 file's path. Then try to play the file outside of Home Assistant it in order to test whether it matches the expected output from the service call parameters.

If the audio is correct, please  test whether Home Assistant can play the file using the `media_player.play_media` service, eg:
```
service: media_player.play_media
data:
  media_content_id: media-source://media_source/local/sounds/temp/chime_tts/b9qrxibw.mp3
  media_content_type: music
target:
  entity_id: media_player.living_speaker
```

If you hear the audio on your media_player speaker then the issue is indeed likely somewhere in the flow of the Chime TTS service call. Please provide the debug logs (again: on the latest version) so that I can hopefully determine the cause of the issue.

**Comment by metbril** [comment]
I will do this tomorrow. For my understanding. What exactly do you refer to with my configuration page?

**Comment by metbril** [comment]
service: chime_tts.say
target:
  entity_id: media_player.living_speaker
data:
  chime_path: bells
  volume_level: 0.3
  message: "Hallo. in ben je virtuele assistent. "
  tts_platform: cloud
  cache: false
  announce: false
  language: no-NL
  options:
    voice: MaartenNeural

**Comment by nimroddolev** [comment]
Hi @metbril. The Chime TTS configuration can be opened from the [integration's page](https://my.home-assistant.io/redirect/integration/?domain=chime_tts)in your Home Assistant. When you open the configuration you can configure file paths and other aspects of Chime TTS.

Were you able to follow the troubleshooting steps from my previous message?

**Comment by metbril** [comment]
Hi, I am testing it at the moment, and I noticed that the generated MP3 has this path:

/config/**None**/media/sounds/temp/chime_tts/xl6gyql2.mp3

    2024-06-23 09:33:31.460 DEBUG (MainThread) [custom_components.chime_tts.helpers.filesystem] File saved to path: /config/None/media/sounds/temp/chime_tts/xl6gyql2.mp3

That 'None' strikes me. Could that be the cause somehow?

**Comment by stfp** [comment]
> /config/None/

I'm also seeing this. (Sorry for the +1 comment, I thought for once it might be useful).

**Comment by nimroddolev** [comment]
Thank you everyone for supplying debug logs and screenshots 🙏

I have just released a new beta version with a fix that hopefully addresses the issue.

I would appreciate it if you could please upgrade to the beta and let me know if the issue is resolved.

To install the beta version:
1. Open the <a href="https://my.home-assistant.io/redirect/hacs_repository/?owner=nimroddolev&repository=chime_tts&category=integration" title="Chime TTS HACS repository" target="_blank">Chime TTS repository in HACS</a>
1. Click on the `⋮` button in the top right
1. Select the `↻ Redownload` option
1. Enable the `Show beta versions` option (it may take a moment to update the list)
1. Select version `v1.1.3-beta1` from the updated version drop-down list
1. Restart Home Assistant

**Comment by metbril** [comment]
> I would appreciate it if you could please upgrade to the beta and let me know if the issue is resolved.

I installed `1.1.3-beta1` and restarted HA. Then tried the developer tools:

```
service: chime_tts.say
target:
  entity_id: media_player.living_speaker
data:
  chime_path: bells
  volume_level: 0.3
  message: "Hallo. Ik ben je virtuele assistent."
  tts_platform: cloud
  cache: true
  announce: true
  language: nl-NL
  options:
    voice: MaartenNeural
```

and it worked. I had the bell and speech played on the speaker.

**Comment by nimroddolev** [comment]
I'm glad to hear that the initial issue is resolved 👍 

Regarding the IP issue, i'm afraid I don't think I will be able to help (I don't own any Sonos speakers), but perhaps you can try posting to the Home Assistant forums. It's likely someone else has run into the same issue and solved it.

I will close this issue once the fix is included in the next stable release.

**Comment by metbril** [comment]
For future reference re. ip bans:

https://github.com/home-assistant/core/issues/110239

https://github.com/home-assistant/core/issues/88714

**Comment by nimroddolev** [comment]
Fixed in v1.1.3
